frontend/test/metabase/scenarios/question/public/question.cy.spec.js

```diff
@@ -0,0 +1,82 @@
+import { restore, visitQuestionAdhoc } from "__support__/e2e/cypress";
+
+const QUESTION_DATA = {
+  name: "15460",
+  dataset_query: {
+    database: 1,
+    native: {
+      query: "SELECT * FROM PEOPLE WHERE {{source}} AND {{birthdate}}",
+      "template-tags": {
+        birthdate: {
+          id: "08c5ea9d-1579-3503-37f1-cbe4d29e6a28",
+          name: "birthdate",
+          "display-name": "Birthdate",
+          type: "dimension",
+          dimension: ["field", 24, null],
+          "widget-type": "date/all-options",
+          default: "past30years",
+        },
+        source: {
+          id: "37eb6fa2-3677-91d3-6be0-c5dd9113c672",
+          name: "source",
+          "display-name": "Source",
+          type: "dimension",
+          dimension: ["field", 26, null],
+          "widget-type": "string/=",
+          default: "Affiliate",
+        },
+      },
+    },
+    type: "native",
+  },
+};
+
+const EXPECTED_QUERY_PARAMS = "?birthdate=past30years&source=Affiliate";
+
+const saveAndShare = () => {
+  cy.icon("share").click();
+  cy.findByPlaceholderText("What is the name of your card?").type("name");
+  cy.button("Save").click();
+};
+
+const enableSharing = () => {
+  cy.findByText("Enable sharing")
+    .parent()
+    .find("a")
+    .click();
+};
+
+const visitPublicURL = () => {
+  cy.findByDisplayValue(/^http:\/\/localhost:4000/)
```

Comment:
I'd avoid hard coding this. You can use baseUrl.

`const BASE_URL = Cypress.config().baseUrl;`

Reply:
I ended up shortening this to just `/^http/`

Could even be `/^h/`.

This was the trickier part of this test, as Cypress was finding another input before the sharing dialog.

So it's just about filtering by the start of the input value.

Reply:
Thank you for looking at the code, Nemanja. Let's keep improving it 🙇